python/mxnet/symbol/symbol.py

```diff
@@ -1544,8 +1544,35 @@ def optimize_for(self, backend, args=None, aux=None, ctx=None, **kwargs):
             raise RuntimeError('Cannot add new aux in optimize_for since aux is None\n' +
                                'Provide a dictionary to the aux argument to optimize_for')
 
-        # return modified symbol
-        return Symbol(out)
+        new_sym = Symbol(out)
+
+        arg_names = self.list_arguments()
+        new_arg_names = new_sym.list_arguments()
+        deleted_arg_names = set([item for item in arg_names
+                                 if item not in set(new_arg_names)])
+
+        if len(deleted_arg_names) > 0:
+            if args is not None:
+                for a_n in deleted_arg_names:
+                    if a_n in args:
+                        args.pop(a_n)
+            else:
+                warnings.warn('optimize_for deleted some argument. \n' +
+                              'Provide a dictionary to the arg argument to optimize_for')
```

Comment:
Can we clarify this error message to something like:
```
A param was deleted during optimization, but no args dictionary was provided. Please ensure that your model weights match the newly optimized model. 
```
Should we print the names of the deleted_arg_names in the message too (or is that overkill)?

Reply:
Thanks for the suggestion! I think that this new message will be sufficient, printing the list of deleted arg names could be potentially very long and flood the stderr